src/test/java/de/medizininformatikinitiative/torch/assertions/ResourceAssert.java

```diff
@@ -0,0 +1,142 @@
+                                                    {
+                                                    "extension": [
+                                                            {
+                                                                "url": "http://hl7.org/fhir/StructureDefinition/data-absent-reason",
+                                                                "valueCode": "masked"
+                                                            }
+                                                        ]
+                                                    }
+                                                    """);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private final IParser fhirParser = FhirContext.forR4().newJsonParser();
+    private final IFhirPath fhirPathEngine = FhirContext.forR4().newFhirPath();
+
+    protected ResourceAssert(Resource resource) {
+        super(resource, ResourceAssert
+                .class);
+    }
+
+    public ListAssert<String> extractChildrenStringsAt(String path) {
+        List<Base> found = fhirPathEngine.evaluate(actual, path, Base.class);
+
+        List<String> parsed = new LinkedList<>();
+        for(Base base : found) {
+            if (base.isPrimitive()) {
+                parsed.add(fhirParser.encodeToString(base));
+            } else {
+                failWithMessage("Expected a primitive value but found %s", base);
+            }
+        }
+
+        return new ListAssert<>(parsed);
+    }
+
+    public ListAssert<JsonNode> extractElementsAt(String path) {
+        ObjectMapper mapper = new ObjectMapper();
+        List<Base> found = fhirPathEngine.evaluate(actual, path, Base.class);
+
+        List<JsonNode> parsed = new LinkedList<>();
+        for(Base base : found) {
+            var encoded = fhirParser.encodeToString(base);
+
+            if (base.isPrimitive()) {
+                parsed.add(mapper.valueToTree(encoded));
+            } else {
+                try {
+                    parsed.add(mapper.readTree(encoded));
+                } catch (JsonProcessingException e) {
+                    failWithMessage("Could not parse value to JsonNode: %s", encoded);
+                }
+            }
+        }
+
+        return new ListAssert<>(parsed);
+    }
+
+    public ListAssert<String> extractTopElementNames() {
+        ObjectMapper mapper = new ObjectMapper();
+        var encoded = fhirParser.encodeResourceToString(actual);
+
+        try {
+            JsonNode obj = mapper.readTree(encoded);
+            return new ListAssert<>(Lists.newArrayList(obj.fieldNames()));
+        } catch (JsonProcessingException e) {
+            failWithMessage("Could not parse value to JsonNode: %s", encoded);
+            return new ListAssert<>(List.of());
+        }
+    }
+
+    public ResourceAssert hasDataAbsentReasonAt(String path) throws JsonProcessingException {
```

Comment:
@bastianschaffer , @alexanderkiel  - it is important that there are different DARs 
=> i would change the function to 
hasDataAbsentReasonAt(String path, String absentReason)